app/helpers/katello/hosts_and_hostgroups_helper.rb

```diff
@@ -302,22 +310,45 @@ def host_checkin_time(host)
 
     private
 
-    def hostgroup_content_facet(hostgroup, param_host)
-      lifecycle_environment_id = hostgroup.inherited_lifecycle_environment_id
-      content_view_id = hostgroup.inherited_content_view_id
+    def decide_content_source_id(host_or_hostgroup, hostgroup)
```

Comment:
I think the verb "decide" is redundant for these method names.  At the end, you have `content_source_id` and some facet attributes being returned, so I'd just describe the returned elements as a noun.

Reply:
Calling the method `content_source_id` is tricky because hosts and hostgroups already have an attribute with that name and I don't want any confusion.  I came up with a slightly more clunky name, but it's a noun so it meets your requirement. let me know if that works..

Reply:
`inherited_or_own_` is definitely clunkier, but the verbosity of it does make a difference in the readability. The whole host/hostgroup workflow is so confusing already.